src/python/pants/util/contextutil.py

```diff
@@ -153,16 +147,22 @@ def signal_handler_as(sig, handler):
 
 
 @contextmanager
-def temporary_dir(root_dir=None, cleanup=True, suffix='', permissions=None, prefix=tempfile.template):
+def temporary_dir(
+  root_dir: Optional[str] = None,
+  cleanup: bool = True,
+  suffix: Optional[str] = None,
```

Comment:
I changed the default of `suffix` to `None` instead of `''`. It behaves the same. `None` better represents the idea of leaving off this arg.